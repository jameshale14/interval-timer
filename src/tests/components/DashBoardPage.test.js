import React from 'react'
import { fireEvent, cleanup } from 'react-testing-library'
import {renderWithRouterAndRedux} from '../utils/render'
import DashBoardPage from '../../components/DashboardPage'

afterEach(cleanup)

test('should render dashboard correctly', () => {
  const { container } = renderWithRouterAndRedux(<DashBoardPage />)
  expect(container).toMatchSnapshot()

})

test('should navigate to new interval page', () => {
  const { getByText, history } = renderWithRouterAndRedux(<DashBoardPage />)
  const leftClick = { button: 0 }
  const newIntervalButton = getByText('New Interval')
  fireEvent.click(newIntervalButton, leftClick)
  expect(history.location.pathname).toEqual('/create')
})