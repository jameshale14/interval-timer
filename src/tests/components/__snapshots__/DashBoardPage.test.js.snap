// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dashboard correctly 1`] = `
<div>
  <div>
    <a
      class="button"
      href="/create"
    >
      New Interval
    </a>
    <div>
      <h1>
        Intervals
      </h1>
      <p>
        No intervals
      </p>
    </div>
  </div>
</div>
`;
